realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -380,7 +380,7 @@ public E remove(int location) {
      * @throws NullPointerException if {@code object} is {@code null}.
      */
     @Override
-    public boolean remove(Object object) {
+    public boolean remove(@Nullable Object object) {
```

Comment:
Does this make sense? Can you remove the `null` element in a standard `ArrayList` ?